modules/eks/ray-cluster/ray_cluster_stack.py

```diff
@@ -0,0 +1,144 @@
+        Tags.of(scope=cast(IConstruct, self)).add(key="Deployment", value=full_dep_mod)
+
+        provider = eks.OpenIdConnectProvider.from_open_id_connect_provider_arn(
+            self, "OIDCProvider", eks_openid_connect_provider_arn
+        )
+
+        eks_cluster = eks.Cluster.from_cluster_attributes(
+            self,
+            "EKSCluster",
+            cluster_name=eks_cluster_name,
+            open_id_connect_provider=provider,
+            kubectl_role_arn=eks_admin_role_arn,
+            kubectl_layer=KubectlV29Layer(self, "Kubectlv29Layer"),
+        )
+
+        eks_cluster.add_helm_chart(
+            "RayCluster",
+            chart="ray-cluster",
+            repository="https://ray-project.github.io/kuberay-helm/",
+            namespace=namespace_name,
+            version="1.1.1",
+            wait=True,
+            values={
+                "image": {
+                    "repository": "rayproject/ray-ml",
+                    "tag": "2.23.0",
+                    "pullPolicy": "IfNotPresent",
+                },
+                "nameOverride": "kuberay",
+                "fullnameOverride": "kuberay",
+                "head": {
+                    "rayVersion": "2.23.0",
+                    "rayStartParams": {"num-cpus": "0"},
+                    "serviceAccountName": service_account_name,
+                    "enableInTreeAutoscaling": enable_autoscaling,
+                    "autoscalerOptions": {
+                        "upscalingMode": "Default",
+                        "idleTimeoutSeconds": autoscaler_idle_timeout_seconds,
+                        # The default Autoscaler resource limits and requests
+                        # should be sufficient for production use-cases.
+                        "resources": {
+                            "limits": {
+                                "cpu": "500m",
+                                "memory": "512Mi",
+                            },
+                            "requests": {
+                                "cpu": "500m",
+                                "memory": "512Mi",
+                            },
+                        },
+                    },
+                    "resources": head_resources,
+                    "volumes": [{"name": "log-volume", "emptyDir": {}}],
+                    "volumeMounts": [{"mountPath": "/tmp/ray", "name": "log-volume"}],
+                },
+                "worker": {
+                    "replicas": worker_replicas,
+                    "minReplicas": worker_min_replicas,
+                    "maxReplicas": worker_max_replicas,
+                    "serviceAccountName": service_account_name,
+                    "resources": worker_resources,
+                    "volumes": [{"name": "log-volume", "emptyDir": {}}],
+                    "volumeMounts": [{"mountPath": "/tmp/ray", "name": "log-volume"}],
+                },
+                "service": {
```

Comment:
if you do not set this service to `ClusterIp`, wouldn't it automatically create a service of ClusterIp for you?

Reply:
Thanks, yes it will create one by default